chapters/inheritance.tex

```diff
@@ -567,6 +567,28 @@ \subsection{Final Element Modification Prevention}\label{final-element-modificat
   PI c1(k = 2, T = 3); // fine, will indirectly change tf.b to 2 * {3, 1}
   PI c2(tf(b = {1}));  // error, b is declared as final
 end Test;
+\end{lstlisting}
+\end{example}
+If a class is final it means that there cannot be modifiers for its elements when using the class.
+\begin{example}
+\begin{lstlisting}[language=modelica]
+model Test2
+  final model MyTransferFunction=TransferFunction(b={1,2});
+  MyTransferFunction tf1;           // fine
+  MyTransferFunction tf2(a={1,2});  // error, the entire class is final
+  model M=MyTransferFunction(a={4});// error, the entire class is final
+  model TFX
+    extends MyTransferFunction;     // fine
+    Real foo = 1.0;
+  end TFX;
+  TFX tfx(foo=2.0);                 // fine
+  TFX tfx2(a={1,3});                // error, a is still final
+  model TFX3=TFX(a={1,4});          // error, a is still final
+
+  A a(redeclare model X=MyTransferFunction); // fine
```

Comment:
And if `A` uses `X` with modifiers, that's an error, I suppose?  (Adding more examples may not be the right remedy, see main review comment.)

Reply:
Obsolete comment.  Resolving.